ted_data_eu/adapters/storage.py

```diff
@@ -0,0 +1,103 @@
+import json
+from typing import Dict, List
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+from ted_data_eu import config
+from ted_data_eu.adapters.storage_abc import DocumentStorageABC
+
+BASIC_HEADERS = {'Content-type': 'application/x-ndjson'}
+
+
+class ElasticStorage(DocumentStorageABC):
+    """
+       Implements interaction with ElasticSearch storage by using its API.
+    """
+
+    def __init__(self,
+                 elastic_index: str,
+                 host: str = None,
+                 user: str = None,
+                 password: str = None):
+        """
+           Implements interaction with ElasticSearch storage by using its API.
+
+            :param elastic_index: ElasticSearch index (schema) where the documents will be stored
+            :param host: Elastic API host and port (if different from the defaults for http and https)
+            :param user: Elastic API user
+            :param password: Elastic API password
+            :return:
+        """
+        self.host = host or config.ELASTIC_HOST
+        self.auth = HTTPBasicAuth(user or config.ELASTIC_USER,
+                                  password or config.ELASTIC_PASSWORD)
+        self.elastic_index = elastic_index
+
+    def add_document(self, document: Dict) -> Dict:
+        """
+           Add document to storage.
+
+            :param document: Document to be stored
+            :return: dict with Elastic API Response
+        """
+        response = requests.post(url=f"{self.host}/{self.elastic_index}/_doc/",
+                                 headers=BASIC_HEADERS,
+                                 auth=self.auth,
+                                 data=json.dumps(document))
+        deserialized_response = json.loads(response.content)
+        if response.status_code != 201:
+            raise Exception(
```

Comment:
Create custom exception, is difficult to catch general Exception, create ElasticStorageExaption class.